### PR TITLE
BOM updates to main

### PR DESCRIPTION
**Pull request by the-scrambler**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -79,6 +80,6 @@ dependencies {
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.0'
```
Comment (osuleymanova) [inline app/build.gradle:80]: 3.5.1- the latest

File: build.gradle [hunk 2]
```diff
@@ -2,15 +2,14 @@
...
-        compose_compiler_version = '1.3.2'
+        compose_compiler_version = '1.4.0'
     }
 }
 
 plugins {
-    id 'com.android.application' version '7.3.1' apply false
-    id 'com.android.library' version '7.3.1' apply false
-    id 'org.jetbrains.kotlin.android' version '1.7.20' apply false
+    id 'com.android.application' version '7.4.0' apply false
```
Comment (osuleymanova) [inline build.gradle]: 7.4.1- the latest

File: app/build.gradle (hunk 1 above)
Comment (the-scrambler) [reply app/build.gradle:80]: If we update this then we have to update the codelab. So I'm focusing mostly on BOM with the exception of other dependencies not mentioned in the codelab to reduce overhead.

Comment (osuleymanova) [reply app/build.gradle:80]: Ok, noted. Thanks!

**Review by yenerm — APPROVED** [review]

**Pull request merged**